### PR TITLE
Update `databricks_pipeline` resource to match latest APIs

### PR DESCRIPTION
**Pull request by alexott**
This change introduces a number of new resources in Pipeline settings, and support for
cloud-specific attributes for clusters

### CODE REVIEW

**Review by nfx — CHANGES_REQUESTED** [review]

File: pipelines/resource_pipeline.go [hunk 1]
```diff
@@ -27,10 +28,12 @@ type pipelineCluster struct {
...
-	NodeTypeID       string                  `json:"node_type_id,omitempty" tf:"group:node_type,computed"`
-	DriverNodeTypeID string                  `json:"driver_node_type_id,omitempty" tf:"conflicts:instance_pool_id,computed"`
-	InstancePoolID   string                  `json:"instance_pool_id,omitempty" tf:"group:node_type"`
-	AwsAttributes    *clusters.AwsAttributes `json:"aws_attributes,omitempty" tf:"conflicts:instance_pool_id"`
+	NodeTypeID           string                  `json:"node_type_id,omitempty" tf:"group:node_type,computed"`
+	DriverNodeTypeID     string                  `json:"driver_node_type_id,omitempty" tf:"computed"`
+	InstancePoolID       string                  `json:"instance_pool_id,omitempty" tf:"group:node_type"`
+	DriverInstancePoolID string                  `json:"driver_instance_pool_id,omitempty"`
+	AwsAttributes        *clusters.AwsAttributes `json:"aws_attributes,omitempty" tf:"suppress_diff"`
+	GcpAttributes        *clusters.GcpAttributes `json:"gcp_attributes,omitempty" tf:"suppress_diff"`
```
[the remark this answers is not in the document]
Comment (alexott) [reply pipelines/resource_pipeline.go:36]: No `azure_attributes` - only AWS & GCP

**Review by nfx — APPROVED** [review]

File: pipelines/resource_pipeline.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (anew) [reply pipelines/resource_pipeline.go:36]: azure attributes are added in https://github.com/databricks/universe/pull/178433

[the remark this answers is not in the document]
Comment (alexott) [reply pipelines/resource_pipeline.go:36]: Ok. There were no yesterday when I checked copied.proto… I’ll add them in follow up PR

File: pipelines/resource_pipeline.go [hunk 2]
```diff
@@ -60,14 +63,18 @@ type filters struct {
 type pipelineSpec struct {
 	ID                  string            `json:"id,omitempty" tf:"computed"`
 	Name                string            `json:"name,omitempty"`
-	Storage             string            `json:"storage,omitempty"`
+	Storage             string            `json:"storage,omitempty" tf:"force_new"`
 	Configuration       map[string]string `json:"configuration,omitempty"`
 	Clusters            []pipelineCluster `json:"clusters,omitempty" tf:"slice_set,alias:cluster"`
 	Libraries           []pipelineLibrary `json:"libraries,omitempty" tf:"slice_set,alias:library"`
-	Filters             *filters          `json:"filters"`
+	Filters             *filters          `json:"filters,omitempty"`
```
Comment (anew) [inline pipelines/resource_pipeline.go:70]: can we not document the filters? They only apply to scala pipelines, which are not supported any more (except for a handful of legacy ones that already existed when we removed scala support).

File: pipelines/resource_pipeline.go [hunk 3]
```diff
@@ -206,18 +213,26 @@ func adjustPipelineResourceSchema(m map[string]*schema.Schema) map[string]*schem
 
 	awsAttributes, _ := clustersSchema["aws_attributes"].Elem.(*schema.Resource)
 	awsAttributesSchema := awsAttributes.Schema
```
Comment (anew) [inline pipelines/resource_pipeline.go:215]: we will now support all aws attributes except EBS volume spec.

File: pipelines/resource_pipeline.go [hunk 4]
```diff
@@ -206,18 +213,26 @@ func adjustPipelineResourceSchema(m map[string]*schema.Schema) map[string]*schem
...
-	delete(awsAttributesSchema, "first_on_demand")
 	delete(awsAttributesSchema, "availability")
 	delete(awsAttributesSchema, "spot_bid_price_percent")
 	delete(awsAttributesSchema, "ebs_volume_type")
 	delete(awsAttributesSchema, "ebs_volume_count")
 	delete(awsAttributesSchema, "ebs_volume_size")
 
+	gcpAttributes, _ := clustersSchema["gcp_attributes"].Elem.(*schema.Resource)
+	gcpAttributesSchema := gcpAttributes.Schema
+	delete(gcpAttributesSchema, "use_preemptible_executors")
```
Comment (anew) [inline pipelines/resource_pipeline.go:224]: we will now support all GCP attribuyes.

File: pipelines/resource_pipeline.go (hunk 2 above)
Comment (alexott) [reply pipelines/resource_pipeline.go:70]: I can remove them completely

**Comment by anew** [comment]
they would be ignored. 

I think it would be best if you wait until this change is rolled out.

File: pipelines/resource_pipeline.go (hunk 2 above)
Comment (nfx) [reply pipelines/resource_pipeline.go:70]: @mgyucht possibly used them for BUI, don't remove without asking him first

**Comment by nfx** [comment]
@alexott no breaking changes allowed! :) just remove from docs.

**Comment by alexott** [comment]
It's already not documented, and this PR removed it from the example...

**Pull request merged**